test/test_individual.py

```diff
@@ -1,139 +1,165 @@
 import math
+import numpy as np
 import pickle
+from collections import namedtuple
+
 import pytest
 
 import cgp
-from cgp.individual import IndividualSingleGenome
+from cgp import IndividualSingleGenome, IndividualMultiGenome
 from cgp.genome import ID_INPUT_NODE, ID_OUTPUT_NODE, ID_NON_CODING_GENE
 
 
-def test_pickle_individual():
+TestParams = namedtuple("TestParams", ["genome_params", "primitives", "dna", "target_function"])
+params_list = [
+    TestParams(
+        genome_params={
+            "n_inputs": 1,
+            "n_outputs": 1,
+            "n_columns": 2,
+            "n_rows": 1,
+            "levels_back": 2,
+        },
+        primitives=(cgp.Add, cgp.Parameter),
+        dna=[
+            ID_INPUT_NODE,
+            ID_NON_CODING_GENE,
+            ID_NON_CODING_GENE,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            ID_OUTPUT_NODE,
+            2,
+            ID_NON_CODING_GENE,
+        ],
+        target_function=lambda x, c: x + c,
+    )
+]
+FuncArgs = namedtuple("FuncArgs", ["x", "c"])
+
+func_args_list = [FuncArgs(x=[3.0, 5.0], c=1.0), FuncArgs(x=[3.0], c=2.0)]
+
+
+def _create_genome(genome_params, primitives, dna):
+    genome = cgp.Genome(**genome_params, primitives=primitives)
+    genome.dna = dna
+    return genome
+
+
+def _create_individual(fitness, genome, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return IndividualSingleGenome(fitness, genome)
+    elif individual_type == "MultiGenome":
+        return IndividualMultiGenome(fitness, [genome])
+    else:
+        raise NotImplementedError
+
+
+def _unpack_evaluation(value, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return value
+    elif individual_type == "MultiGenome":
+        return value[0]
+    else:
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError()
```
it think you need to call the exception, don't you?

Reply:
No, `NotImplementedError` is a shortcut for `NotImplementedError()`. We just need to call it if we want to print a message.

Reply:
cool, i learned something! :) but wouldn't it make sense to provide a message? ;)

Reply:
Fair point. Added a message. 27ff50681435a44ab24f87dc6efef799c55d26a3